### PR TITLE
Init gulp starter kit template, added data and data access layers

### PR DESCRIPTION
**Pull request by MiletaStanisic**
*Note
ScrumFire tasks for data & data access layers

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,10 +1,78 @@
...
-## Note:
-This is just the plugin source code. if you need the full BuildFire SDK you can find it here https://github.com/BuildFire/sdk
+## run
+then when ready for final test and release run `npm start` or `gulp build`
+This will prodice a sister folder with `_release` as a postfix on teh root folder
+thest his optomized copy then zip and publish
 
-### Special Thanks to 
-openweathermap.org for providing the open api
+## merge
```
Comment (denza) [inline README.md]: Please make sure to keep all changes from previous version.

File: control/content/samples/searchTable/dummyData.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
+
```
Comment (denza) [inline control/content/samples/searchTable/dummyData.js]: Remove this file

File: control/content/samples/searchTable/searchTable.css [hunk 3]
```diff
@@ -0,0 +1,47 @@
+thead, tbody tr {
```
Comment (denza) [inline control/content/samples/searchTable/searchTable.css]: Remove this file

File: control/content/samples/searchTable/searchTable.html [hunk 4]
```diff
@@ -0,0 +1,89 @@
+<!DOCTYPE html>
```
Comment (denza) [inline control/content/samples/searchTable/searchTable.html]: Remove this file

File: control/content/samples/searchTable/searchTableConfig.js [hunk 5]
```diff
@@ -0,0 +1,45 @@
+
```
Comment (denza) [inline control/content/samples/searchTable/searchTableConfig.js]: Remove this file

File: control/content/samples/searchTable/searchTableHelper.js [hunk 6]
```diff
@@ -0,0 +1,221 @@
+class SearchTableHelper{
```
Comment (denza) [inline control/content/samples/searchTable/searchTableHelper.js]: Remove this file

File: control/content/samples/sortableList/sortableList.css [hunk 7]
```diff
@@ -0,0 +1,124 @@
+/* Draggable List  */
```
Comment (denza) [inline control/content/samples/sortableList/sortableList.css]: Remove this file

File: control/content/samples/sortableList/sortableList.html [hunk 8]
```diff
@@ -0,0 +1,105 @@
+<!DOCTYPE html>
```
Comment (denza) [inline control/content/samples/sortableList/sortableList.html]: Remove this file

File: control/content/samples/sortableList/sortableList.js [hunk 9]
```diff
@@ -0,0 +1,184 @@
+﻿
```
Comment (denza) [inline control/content/samples/sortableList/sortableList.js]: Remove this file

File: control/content/samples/sortableList/sortableListUI.js [hunk 10]
```diff
@@ -0,0 +1,116 @@
+const sortableListUI= {
```
Comment (denza) [inline control/content/samples/sortableList/sortableListUI.js]: Remove this file

File: control/content/samples/subPage/index.html [hunk 11]
```diff
@@ -0,0 +1,58 @@
+<!DOCTYPE html>
```
Comment (denza) [inline control/content/samples/subPage/index.html]: Remove this file

File: control/content/samples/subPage/subPage.css [hunk 12]
```diff
@@ -0,0 +1,57 @@
+.subPage{
```
Comment (denza) [inline control/content/samples/subPage/subPage.css]: Remove this file

File: control/content/index.html [hunk 13]
```diff
@@ -2,54 +2,25 @@
...
+     <script src="../../dataAccess/settings.js"></script>
+     <script src="../../dataAccess/weatherData.js"></script>
+     <!-- End of Scripts -->
 
-    load();
 
+</head>
+<body>
+<script>
+    //location = "samples/sortableList/sortableList.html";
```
Comment (denza) [inline control/content/index.html]: Please remove commented code

**Review by denza — COMMENTED** [review]
Looks good

File: tests/basic/settings.test.js [hunk 14]
```diff
@@ -0,0 +1,83 @@
+class SettingsTest {
+    /**
+     * Total number of tests
+     */
+    static get testCount() {
+      return 4;
```
Comment (denza) [inline tests/basic/settings.test.js]: testCount should not be used if you have only one instance.

File: dataAccess/weatherData.js [hunk 15]
```diff
@@ -12,6 +12,32 @@ class WeatherData {
             return callback(records);
         });
     };
+  
+    /**
+    * Gets all barcodes
```
Comment (denza) [inline dataAccess/weatherData.js]: Barcodes?

File: dataAccess/weatherData.js [hunk 16]
```diff
@@ -34,7 +60,6 @@ class WeatherData {
     static add = (data, callback) => {
         data.createdBy = authManager.currentUser.email;
```
Comment (denza) [inline dataAccess/weatherData.js]: Please use userId, not email address.

File: dataAccess/weatherData.js [hunk 17]
```diff
@@ -12,6 +12,32 @@ class WeatherData {
...
+    */
+    static get = (callback) => {
+      buildfire.datastore.get(WeatherData.TAG, (error, record) => {
+        if (error) return callback(error);
+
+        if (!record.data.place) {
+          const cmd = {
+            latestUpdate: new Date(),
+            createdOn: new Date(),
+            createdBy: authManager.currentUser.email
```
Comment (denza) [inline dataAccess/weatherData.js]: Please use userId, not email address.

File: dataAccess/weatherData.js [hunk 18]
```diff
@@ -48,10 +73,15 @@ class WeatherData {
...
-    static set = (data, callback) => {
-        data.lastUpdatedOn = new Date();
-        data.lastUpdatedBy = authManager.currentUser.email;
-        buildfire.publicData.update(data.id, data, WeatherData.TAG, (error, record) => {
+      static set = (data, callback) => {
+        const cmd = {
+            $set: {
+              place: data.latestUpdate,
+              lastUpdatedOn: new Date(),
+              lastUpdatedBy: authManager.currentUser.email
```
Comment (denza) [inline dataAccess/weatherData.js]: Please use userId, not email address.

File: tests/basic/weather.test.js [hunk 19]
```diff
@@ -0,0 +1,64 @@
...
+      Settings.set(new WeatherData(this.initial), () => {});
+    }
+    initial = {};
+    
+    static run = () => {
+      
+      test("Get weather data", () => {
+        WeatherData.get((err, data) => {
+          this.initial = data.latestUpdate;
+          console.log(data);
```
Comment (denza) [inline tests/basic/weather.test.js]: Remove log

File: tests/basic/weather.test.js [hunk 20]
```diff
@@ -0,0 +1,64 @@
...
+
+          if (err) {
+            console.error(err);
```
Comment (denza) [inline tests/basic/weather.test.js]: Remove log

File: tests/basic/weather.test.js [hunk 21]
```diff
@@ -0,0 +1,64 @@
+class WeatherTest {
+  
+    /**
+     * Handle finishing of each test and restore data if it is the last one
+     */
+    static finishTest() {
+      Settings.set(new WeatherData(this.initial), () => {});
+    }
```
Comment (denza) [inline tests/basic/weather.test.js]: We do not need this method

**Comment by najiatallah** [comment]
Approved

File: widget/index.html [hunk 22]
```diff
@@ -2,66 +2,218 @@
...
+    ).then(function (response) {
+        createWeatherImage(response.url);
+    }).catch(function () {
+        console.log('failed to load image');
+    });
+}
+
+fetchWeatherData = (latitude, longitude, tempUnit, apiKey, callback) => {
+    /// Call open weather API to pull data
+    fetch("https://api.openweathermap.org/data/2.5/find?appid=" + apiKey + "&lat=" + latitude + "&lon=" + longitude + tempUnit
```
Comment (mikailbayram) [inline widget/index.html:115]: lets move the endpoints to a higher level variable so that any change in the endpoint wont be "painful" on our side

**Review by mikailbayram — CHANGES_REQUESTED** [review]

**Pull request merged**